File: dataset/output-bk/MISTRAL_codestral-latest/Task194_MISTRAL_codestral-latest.java
// Java
public class Solution {
    public void findSecretWord(String[] words, Master master) {
        for (int i = 0; i < 10; i++) {
            int guessIndex = (int) (Math.random() * words.length);
            String guess = words[guessIndex];
            int matches = master.guess(guess);
            if (matches == 6) {
                System.out.println("You guessed the secret word correctly.");
                return;
            }
            List<String> newWords = new ArrayList<>();
            for (String word : words) {
                if (match(word, guess, matches)) {
                    newWords.add(word);
                }
            }
            words = newWords.toArray(new String[0]);
        }
        System.out.println("Either you took too many guesses, or you did not find the secret word.");
    }

    private boolean match(String word, String guess, int matches) {
        int count = 0;
        for (int i = 0; i < word.length(); i++) {
            if (word.charAt(i) == guess.charAt(i)) {
                count++;
            }
        }
        return count == matches;
    }
}

File: dataset/output-bk/MISTRAL_codestral-latest/Task194_MISTRAL_codestral-latest.c
// C
#include <stdlib.h>
#include <string.h>
#include <stdio.h>

int match(char* word, char* guess, int matches) {
    int count = 0;
    for (int i = 0; i < strlen(word); i++) {
        if (word[i] == guess[i]) {
            count++;
        }
    }
    return count == matches;
}

void findSecretWord(char** words, int wordsSize, Master* master) {
    for (int i = 0; i < 10; i++) {
        int guessIndex = rand() % wordsSize;
        char* guess = words[guessIndex];
        int matches = master->guess(guess);
        if (matches == 6) {
            printf("You guessed the secret word correctly.
");
            return;
        }
        char** newWords = malloc(wordsSize * sizeof(char*));
        int newWordsSize = 0;
        for (int j = 0; j < wordsSize; j++) {
            if (match(words[j], guess, matches)) {
                newWords[newWordsSize++] = words[j];
            }
        }
        words = newWords;
        wordsSize = newWordsSize;
    }
    printf("Either you took too many guesses, or you did not find the secret word.
");
}